src/main/java/bot/telegram/handlers/MessageUpdateHandler.java

```diff
@@ -14,60 +18,58 @@
 @RequiredArgsConstructor
 @Service
 @Slf4j
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
 public class MessageUpdateHandler {
-    private final MainMenuKeyboard menu;
-    private final StateController controller;
-
+    MainMenuKeyboard menu;
+    StateProcessor controller;
+    UserService service;
 
     public BotApiMethod<?> handleMessage(Message message) {
-        BotState botState;
+
         String chatId = String.valueOf(message.getChatId());
+        User user = service.initializeUser(chatId);
+        BotState botState;
+
 
         if (!message.hasText()) {
             return sendError(chatId);
         }
         String input = message.getText();
```

Comment:
@temalead In general, it is not a good thing to use `switch/case` code structure. The utilization of it breaks the open/closed SOLID principal almost instanteneously. As an alternative, you can declare a `java.util.List` of possible commands inside `BotState` enum, and then extract `BotState` by the input command. See below:

```
public enum BotState {
    SHOW_START_MENU(List.of("/start"))

    @Getter
    private final List<STtring> commands;

    BotState(List<String> commands) {
         this.commands = commands;
    }

    // Take a close look here
    public BotState getByCommand(String command) {
         Arrays.stream(values()).filter(it -> it.getCommands().contains(command)).findFirst().orElse(null);
    }
}
```

If you ill do this, switch will be no longer required. @temalead Please, consider this improvement before merging this PR.

Reply:
OMG! It s genius. Thanks a lot. I appreciate your help!
Of course switch case breaks OCP, but ,maybe i`m wrong, i think i can use switch case where only setting bot state- performing only one action.
```
switch (input) { 
    case "/start":
        botState = BotState.GET_START_MENU;
        break;
    //case ...
user.setState(botState);
```
I think, i can save this piece of code. Isn`t it?